core-web/libs/ui/src/lib/dot-tab-buttons/dot-tab-buttons.component.spec.ts

```diff
@@ -13,98 +16,188 @@ import { DotTabButtonsComponent } from './dot-tab-buttons.component';
+    let editID: string;
+    let previewID: string;
+
     const messageServiceMock = new MockDotMessageService({
         'editpage.toolbar.edit.page.clipboard': 'Edit Page',
         'editpage.toolbar.preview.page.clipboard': 'Preview Page'
     });
+
     const createComponent = createComponentFactory({
         component: DotTabButtonsComponent,
         providers: [
             HttpClientTestingModule,
             DotMessagePipe,
             { provide: DotMessageService, useValue: messageServiceMock }
-        ]
+        ],
+        imports: [NgFor, ButtonModule, NgIf, NgClass, TooltipModule, DotMessagePipe]
     });
+
     const optionsMock: SelectItem[] = [
-        { label: 'Edit', value: DotPageMode.EDIT },
-        { label: 'Preview', value: DotPageMode.PREVIEW }
+        { label: 'Edit', value: { id: DotPageMode.EDIT, showDropdownButton: false } },
+        { label: 'Preview', value: { id: DotPageMode.PREVIEW, showDropdownButton: true } }
     ];
 
     beforeEach(() => {
         spectator = createComponent({
             props: {
                 options: optionsMock,
-                mode: DotPageMode.PREVIEW,
-                icon: 'pi-angle-down'
+                activeId: DotPageMode.PREVIEW
             }
         });
+
+        editID = spectator.component._options[0].value.id;
+        previewID = spectator.component._options[1].value.id;
     });
 
     it('should render options', () => {
         const buttons = spectator.queryAll(byTestId('dot-tab-button-text'));
-        expect(spectator.query(byTestId('dot-tab-button-text'))).toBeDefined();
+        expect(buttons[0]).toBeDefined();
         expect(buttons.length).toEqual(2);
         buttons.forEach((button, index) => {
             expect(button.textContent.trim()).toEqual(optionsMock[index].label);
         });
     });
 
     it('should emit openMenu event when showMenu is called', () => {
-        const openMenuSpy = spyOn(spectator.component.dropdownClick, 'emit');
-        spectator.component.showMenu(null);
-        expect(openMenuSpy).toHaveBeenCalled();
+        const openMenuSpy = spyOn(spectator.component.openMenu, 'emit');
+        spectator.component.onClickDropdown(pointerEvent, previewID);
+        expect(openMenuSpy).toHaveBeenCalledWith({
+            event: pointerEvent,
+            menuId: previewID
+        });
+    });
+
+    it('should emit openMenu event when showMenu is called', () => {
+        const openMenuSpy = spyOn(spectator.component.openMenu, 'emit');
+        spectator.component.onClickDropdown(pointerEvent, previewID);
+        expect(openMenuSpy).toHaveBeenCalledWith({
+            event: pointerEvent,
+            menuId: previewID
+        });
+    });
+
+    it('should not emit openMenu event when showMenu is called and the option does not have showDropdownButton setted to true', () => {
+        const openMenuSpy = spyOn(spectator.component.openMenu, 'emit');
+        spectator.component.onClickDropdown(pointerEvent, editID);
```

Comment:
https://ngneat.github.io/spectator/docs/events/#custom-events

Reply:
In this case I cannot use the `triggerEventHandler` because I'm testing a case where you call the function and the dropdown does not exists in the DOM